src/ui/ons/ONSTextField.svelte

```diff
@@ -1,15 +1,56 @@
 <script>
-  export let id;
-  export let value = "";
+  export let id, type, textFieldValue, onChange;
   export let hint = "";
+  export let inputLabel = "";
+  export let showPlaceholder = false;
+  export let errorMessage = "Custom error message";
+  export let renderErrorMsg = false;
+
+  function typeAction(node) {
+    node.type = type;
+  }
 </script>
 
-<div class="ons-field">
-  <label class="ons-label" for={id}><slot /></label>
-  {#if hint}
-    <span id="description-hint" class="ons-label__description  ons-input--with-description">
-      {hint}
-    </span>
-  {/if}
-  <input type="text" {id} class="ons-input ons-input--text ons-input-type__input" bind:value />
-</div>
+{#if renderErrorMsg}
+  <div class="ons-panel ons-panel--error ons-panel--no-title ons-u-mb-s" id="{id}-error">
+    <span class="ons-u-vh">Error: </span>
+    <div class="ons-panel__body">
+      <p class="ons-panel__error">
+        <strong>{errorMessage}</strong>
+      </p>
+      <div class="ons-field">
+        <label class="ons-label {showPlaceholder ? 'ons-label--placeholder' : ''}" for={id}>{inputLabel}</label>
+        {#if hint}
+          <span id="description-hint" class="ons-label__description  ons-input--with-description">
+            {hint}
+          </span>
+        {/if}
+        <input
+          {id}
+          class="ons-input ons-input--text ons-input-type__input {showPlaceholder ? 'ons-input--placeholder' : ''}"
+          placeholder={showPlaceholder ? inputLabel : ""}
+          bind:value={textFieldValue}
+          on:input={() => onChange(textFieldValue)}
```

Comment:
Good idea to implement `on:input` but when you do so you should call the export `onInput` too to avoid confusion. You could also implement `onChange`

Reply:
Here I took your advice and I renamed the onChange prop to onInput prop so that it follows the event name.I have also added an onChange handler  which uses the on:change event. To see the change check the browser console and hit enter after you type something in the input box, or click outside of the input box.